{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Single Cell Sequencing\n",
    "## Creates h5ad files \n",
    "\n",
    "\n",
    "Uses exon data extracted from: zUMIs_output\\expression\\*.dgecounts.rds and gene data from: zUMIs_output\\expression\\*.gene_names.txt<br>\n",
    "Creates h5ad file from anndata objects. Each file represents data from one plate (sample)<br>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pyreadr in /home/szczotlab/anaconda3/envs/single_cell_sequencing/lib/python3.12/site-packages (0.5.2)\n",
      "Requirement already satisfied: pandas>=1.2.0 in /home/szczotlab/anaconda3/envs/single_cell_sequencing/lib/python3.12/site-packages (from pyreadr) (1.5.3)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in /home/szczotlab/anaconda3/envs/single_cell_sequencing/lib/python3.12/site-packages (from pandas>=1.2.0->pyreadr) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/szczotlab/anaconda3/envs/single_cell_sequencing/lib/python3.12/site-packages (from pandas>=1.2.0->pyreadr) (2024.1)\n",
      "Requirement already satisfied: numpy>=1.21.0 in /home/szczotlab/anaconda3/envs/single_cell_sequencing/lib/python3.12/site-packages (from pandas>=1.2.0->pyreadr) (1.26.4)\n",
      "Requirement already satisfied: six>=1.5 in /home/szczotlab/anaconda3/envs/single_cell_sequencing/lib/python3.12/site-packages (from python-dateutil>=2.8.1->pandas>=1.2.0->pyreadr) (1.16.0)\n"
     ]
    }
   ],
   "source": [
    "# !pip install --quiet anndata\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import anndata as ad\n",
    "!pip install pyreadr\n",
    "import pyreadr\n",
    "import config as general_paths\n",
    "import os\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_path = general_paths.figures_path\n",
    "data_path = general_paths.data_path\n",
    "output_path = general_paths.output_path\n",
    "local_path = general_paths.local_path\n",
    "rds_file_sample1 = os.path.join(local_path, \"alignment/sample1.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample1 = os.path.join(local_path, \"alignment/sample1.h5ad\")  # the file that will be used for Melisas script\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "rds_file_sample1 = os.path.join(local_path, \"alignment/sample1.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample1 = os.path.join(local_path, \"alignment/sample1.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample2 = os.path.join(local_path, \"alignment/sample2.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample2 = os.path.join(local_path, \"alignment/sample2.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample3 = os.path.join(local_path, \"alignment/sample3.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample3 = os.path.join(local_path, \"alignment/sample3.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample4 = os.path.join(local_path, \"alignment/sample4.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample4 = os.path.join(local_path, \"alignment/sample4.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample5 = os.path.join(local_path, \"alignment/sample5.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample5 = os.path.join(local_path, \"alignment/sample5.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample6 = os.path.join(local_path, \"alignment/sample6.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample6 = os.path.join(local_path, \"alignment/sample6.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample7 = os.path.join(local_path, \"alignment/sample7.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample7 = os.path.join(local_path, \"alignment/sample7.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample8 = os.path.join(local_path, \"alignment/sample8.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample8 = os.path.join(local_path, \"alignment/sample8.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample9 = os.path.join(local_path, \"alignment/sample9.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample9 = os.path.join(local_path, \"alignment/sample9.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample10 = os.path.join(local_path, \"alignment/sample10.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample10 = os.path.join(local_path, \"alignment/sample10.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "rds_file_sample11 = os.path.join(local_path, \"alignment/sample11.rds\") # the file read and prepared in R\n",
    "h5ad_file_sample11 = os.path.join(local_path, \"alignment/sample11.h5ad\")  # the file that will be used for Melisas script\n",
    "\n",
    "# file with gene ids and names from kallisto\n",
    "t2g_file_path = os.path.join(local_path, \"t2g.txt\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the RDS file and convert to a DataFrame\n",
    "rds_sample1 = pyreadr.read_r(rds_file_sample1)\n",
    "rds_sample2 = pyreadr.read_r(rds_file_sample2)\n",
    "rds_sample3 = pyreadr.read_r(rds_file_sample3)\n",
    "rds_sample4 = pyreadr.read_r(rds_file_sample4)\n",
    "rds_sample5 = pyreadr.read_r(rds_file_sample5)\n",
    "rds_sample6 = pyreadr.read_r(rds_file_sample6)\n",
    "rds_sample7 = pyreadr.read_r(rds_file_sample7)\n",
    "rds_sample8 = pyreadr.read_r(rds_file_sample8)\n",
    "rds_sample9 = pyreadr.read_r(rds_file_sample9)\n",
    "rds_sample10 = pyreadr.read_r(rds_file_sample10)\n",
    "rds_sample11 = pyreadr.read_r(rds_file_sample11)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(28914, 384)\n"
     ]
    }
   ],
   "source": [
    "# convert to dataframes (they are different sizes)\n",
    "rds_sample1_df = rds_sample1[list(rds_sample1.keys())[0]]\n",
    "rds_sample2_df = rds_sample2[list(rds_sample2.keys())[0]]\n",
    "rds_sample3_df = rds_sample3[list(rds_sample3.keys())[0]]\n",
    "rds_sample4_df = rds_sample4[list(rds_sample4.keys())[0]]\n",
    "rds_sample5_df = rds_sample5[list(rds_sample5.keys())[0]]\n",
    "rds_sample6_df = rds_sample6[list(rds_sample6.keys())[0]]\n",
    "rds_sample7_df = rds_sample7[list(rds_sample7.keys())[0]]\n",
    "rds_sample8_df = rds_sample8[list(rds_sample8.keys())[0]]\n",
    "rds_sample9_df = rds_sample9[list(rds_sample9.keys())[0]]\n",
    "rds_sample10_df = rds_sample10[list(rds_sample10.keys())[0]]\n",
    "rds_sample11_df = rds_sample11[list(rds_sample11.keys())[0]]\n",
    "print(rds_sample11_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AACCATCGGCAACTACCACT</th>\n",
       "      <th>AACCATCGGCAAGTTATCGG</th>\n",
       "      <th>AACCATCGGCCACAATCCAC</th>\n",
       "      <th>AACCATCGGCCACTAACCGG</th>\n",
       "      <th>AACCATCGGCCCGTTCGTAT</th>\n",
       "      <th>AACCATCGGCCGACGGTTGT</th>\n",
       "      <th>AACCATCGGCCGAGTCATTA</th>\n",
       "      <th>AACCATCGGCGAAGCTAGCC</th>\n",
       "      <th>AACCATCGGCGTAACGTCTA</th>\n",
       "      <th>AACCATCGGCTCCAATGCAA</th>\n",
       "      <th>...</th>\n",
       "      <th>TTGTCTGCAAGCACAGTATT</th>\n",
       "      <th>TTGTCTGCAATACAACTAGC</th>\n",
       "      <th>TTGTCTGCAATAGGTGGCGA</th>\n",
       "      <th>TTGTCTGCAATAGTTAGGCT</th>\n",
       "      <th>TTGTCTGCAATCTGCAACAA</th>\n",
       "      <th>TTGTCTGCAATGCATCATAC</th>\n",
       "      <th>TTGTCTGCAATGGCCTTAAT</th>\n",
       "      <th>TTGTCTGCAATTACAACACG</th>\n",
       "      <th>TTGTCTGCAATTCAGGAGAT</th>\n",
       "      <th>TTGTCTGCAATTCTCTGATC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rownames</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0610005C13Rik</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610009E02Rik</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610009L18Rik</th>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610010K14Rik</th>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610025J13Rik</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Trav6-7-dv9</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Usp9y</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vmn1r80</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Wfdc11</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Zfp36l3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35546 rows Ã— 4224 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               AACCATCGGCAACTACCACT  AACCATCGGCAAGTTATCGG  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   0.0   \n",
       "0610009L18Rik                   3.0                   0.0   \n",
       "0610010K14Rik                   2.0                   2.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               AACCATCGGCCACAATCCAC  AACCATCGGCCACTAACCGG  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   0.0   \n",
       "0610009L18Rik                   2.0                   0.0   \n",
       "0610010K14Rik                   4.0                   0.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               AACCATCGGCCCGTTCGTAT  AACCATCGGCCGACGGTTGT  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   0.0   \n",
       "0610009L18Rik                   1.0                   2.0   \n",
       "0610010K14Rik                   4.0                   3.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               AACCATCGGCCGAGTCATTA  AACCATCGGCGAAGCTAGCC  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   1.0   \n",
       "0610009L18Rik                   0.0                   0.0   \n",
       "0610010K14Rik                   1.0                   0.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               AACCATCGGCGTAACGTCTA  AACCATCGGCTCCAATGCAA  ...  \\\n",
       "rownames                                                   ...   \n",
       "0610005C13Rik                   0.0                   0.0  ...   \n",
       "0610009E02Rik                   0.0                   0.0  ...   \n",
       "0610009L18Rik                   0.0                   0.0  ...   \n",
       "0610010K14Rik                   3.0                   3.0  ...   \n",
       "0610025J13Rik                   0.0                   0.0  ...   \n",
       "...                             ...                   ...  ...   \n",
       "Trav6-7-dv9                     0.0                   0.0  ...   \n",
       "Usp9y                           0.0                   0.0  ...   \n",
       "Vmn1r80                         0.0                   0.0  ...   \n",
       "Wfdc11                          0.0                   0.0  ...   \n",
       "Zfp36l3                         0.0                   0.0  ...   \n",
       "\n",
       "               TTGTCTGCAAGCACAGTATT  TTGTCTGCAATACAACTAGC  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   0.0   \n",
       "0610009L18Rik                   0.0                   0.0   \n",
       "0610010K14Rik                   2.0                   4.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               TTGTCTGCAATAGGTGGCGA  TTGTCTGCAATAGTTAGGCT  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   0.0   \n",
       "0610009L18Rik                   3.0                   0.0   \n",
       "0610010K14Rik                   2.0                   8.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               TTGTCTGCAATCTGCAACAA  TTGTCTGCAATGCATCATAC  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   0.0   \n",
       "0610009L18Rik                   1.0                   0.0   \n",
       "0610010K14Rik                  10.0                   5.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               TTGTCTGCAATGGCCTTAAT  TTGTCTGCAATTACAACACG  \\\n",
       "rownames                                                    \n",
       "0610005C13Rik                   0.0                   0.0   \n",
       "0610009E02Rik                   0.0                   0.0   \n",
       "0610009L18Rik                   2.0                   0.0   \n",
       "0610010K14Rik                   1.0                   6.0   \n",
       "0610025J13Rik                   0.0                   0.0   \n",
       "...                             ...                   ...   \n",
       "Trav6-7-dv9                     0.0                   0.0   \n",
       "Usp9y                           0.0                   0.0   \n",
       "Vmn1r80                         0.0                   0.0   \n",
       "Wfdc11                          0.0                   0.0   \n",
       "Zfp36l3                         0.0                   0.0   \n",
       "\n",
       "               TTGTCTGCAATTCAGGAGAT  TTGTCTGCAATTCTCTGATC  \n",
       "rownames                                                   \n",
       "0610005C13Rik                   0.0                   0.0  \n",
       "0610009E02Rik                   0.0                   1.0  \n",
       "0610009L18Rik                   0.0                   1.0  \n",
       "0610010K14Rik                   5.0                   3.0  \n",
       "0610025J13Rik                   0.0                   1.0  \n",
       "...                             ...                   ...  \n",
       "Trav6-7-dv9                     0.0                   0.0  \n",
       "Usp9y                           0.0                   0.0  \n",
       "Vmn1r80                         0.0                   0.0  \n",
       "Wfdc11                          0.0                   0.0  \n",
       "Zfp36l3                         0.0                   0.0  \n",
       "\n",
       "[35546 rows x 4224 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge whole dataframe, so that all will have the same amount of genes\n",
    "unified_df = pd.concat([rds_sample1_df,rds_sample2_df,rds_sample3_df,rds_sample4_df,rds_sample5_df,rds_sample6_df,rds_sample7_df,rds_sample8_df,rds_sample9_df,rds_sample10_df,rds_sample11_df],axis=1)\n",
    "# replace all NaN values with zero counts\n",
    "unified_df = unified_df.fillna(0.0)\n",
    "unified_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# t2g = pd.read_csv(t2g_file_path, header=None, names=[\"tid\", \"gene_id\", \"gene_name\"], sep=\"\\t\")\n",
    "# t2g.index = t2g.gene_id### FOR THAT WE USE T2G VARIABLE THAT OPENS T2G FILE (FILE WITH LINK BETWEEN GENE CODE AND GENE NAME)\n",
    "\n",
    "# t2g = t2g.loc[~t2g.index.duplicated(keep='first')]\n",
    "# gene_ids_kallisto = t2g['gene_id'].tolist()\n",
    "# gene_ids_scilifelab = list(unified_df.index)\n",
    "# diffelence1 = list(set(gene_ids_kallisto) - set(gene_ids_scilifelab))\n",
    "# diffelence2 = list(set(gene_ids_scilifelab) - set(gene_ids_kallisto))\n",
    "# print(f\"Gene ids that are in kallisto file but not in scilifelab: {len(diffelence1)}\\nGene ids that are in scilifelab but not in kallisto: {len(diffelence2)}\") \n",
    "# # there are 308 gene ids from scilifelab that are not in kallisto ids!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# split back to original dataframes but same numbers of genes now\n",
    "# check how many cells were in each of the data sets\n",
    "total_barcodes_df1 = rds_sample1_df.shape[1]\n",
    "total_barcodes_df2 = rds_sample2_df.shape[1]\n",
    "total_barcodes_df3 = rds_sample3_df.shape[1]\n",
    "total_barcodes_df4 = rds_sample4_df.shape[1]\n",
    "total_barcodes_df5 = rds_sample5_df.shape[1]\n",
    "total_barcodes_df6 = rds_sample6_df.shape[1]\n",
    "total_barcodes_df7 = rds_sample7_df.shape[1]\n",
    "total_barcodes_df8 = rds_sample8_df.shape[1]\n",
    "total_barcodes_df9 = rds_sample9_df.shape[1]\n",
    "total_barcodes_df10 = rds_sample10_df.shape[1]\n",
    "total_barcodes_df11 = rds_sample11_df.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(384, 384, 384, 384, 384, 384, 384, 384, 384, 384, 384)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rds_sample1_df = unified_df.iloc[:,0:total_barcodes_df1]\n",
    "till_col = total_barcodes_df1+total_barcodes_df2\n",
    "rds_sample2_df = unified_df.iloc[:,total_barcodes_df1:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df3\n",
    "rds_sample3_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df4\n",
    "rds_sample4_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df5\n",
    "rds_sample5_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df6\n",
    "rds_sample6_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df7\n",
    "rds_sample7_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df8\n",
    "rds_sample8_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df9\n",
    "rds_sample9_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df10\n",
    "rds_sample10_df = unified_df.iloc[:,from_col:till_col]\n",
    "from_col = till_col\n",
    "till_col = from_col+total_barcodes_df11\n",
    "rds_sample11_df = unified_df.iloc[:,from_col:till_col]\n",
    "\n",
    "# verify number of cells in each new dataframe\n",
    "rds_sample1_df.shape[1],rds_sample2_df.shape[1],rds_sample3_df.shape[1],rds_sample4_df.shape[1],rds_sample5_df.shape[1],rds_sample6_df.shape[1],rds_sample7_df.shape[1],rds_sample8_df.shape[1],rds_sample9_df.shape[1],rds_sample10_df.shape[1],rds_sample11_df.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# transpose before creating anndata object, so that gene ids are columns and barcodes are rows\n",
    "transposed_rds_sample1_df = rds_sample1_df.transpose()# Create an AnnData object\n",
    "adata1 = ad.AnnData(\n",
    "    X=transposed_rds_sample1_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample1_df.index},\n",
    "    var={'gene_id': transposed_rds_sample1_df.columns}  \n",
    ")\n",
    "transposed_rds_sample2_df = rds_sample2_df.transpose()# Create an AnnData object\n",
    "adata2 = ad.AnnData(\n",
    "    X=transposed_rds_sample2_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample2_df.index},\n",
    "    var={'gene_id': transposed_rds_sample2_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample3_df = rds_sample3_df.transpose()# Create an AnnData object\n",
    "adata3 = ad.AnnData(\n",
    "    X=transposed_rds_sample3_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample3_df.index},\n",
    "    var={'gene_id': transposed_rds_sample3_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample4_df = rds_sample4_df.transpose()# Create an AnnData object\n",
    "adata4 = ad.AnnData(\n",
    "    X=transposed_rds_sample4_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample4_df.index},\n",
    "    var={'gene_id': transposed_rds_sample4_df.columns}  \n",
    ")\n",
    "transposed_rds_sample5_df = rds_sample5_df.transpose()# Create an AnnData object\n",
    "adata5 = ad.AnnData(\n",
    "    X=transposed_rds_sample5_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample5_df.index},\n",
    "    var={'gene_id': transposed_rds_sample5_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample6_df = rds_sample6_df.transpose()# Create an AnnData object\n",
    "adata6 = ad.AnnData(\n",
    "    X=transposed_rds_sample6_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample6_df.index},\n",
    "    var={'gene_id': transposed_rds_sample6_df.columns}  \n",
    ")\n",
    "\n",
    "transposed_rds_sample7_df = rds_sample7_df.transpose()# Create an AnnData object\n",
    "adata7 = ad.AnnData(\n",
    "    X=transposed_rds_sample7_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample7_df.index},\n",
    "    var={'gene_id': transposed_rds_sample7_df.columns}  \n",
    ")\n",
    "transposed_rds_sample8_df = rds_sample8_df.transpose()# Create an AnnData object\n",
    "adata8 = ad.AnnData(\n",
    "    X=transposed_rds_sample8_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample8_df.index},\n",
    "    var={'gene_id': transposed_rds_sample8_df.columns}  \n",
    ")\n",
    "transposed_rds_sample9_df = rds_sample9_df.transpose()# Create an AnnData object\n",
    "adata9 = ad.AnnData(\n",
    "    X=transposed_rds_sample9_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample9_df.index},\n",
    "    var={'gene_id': transposed_rds_sample9_df.columns}  \n",
    ")\n",
    "\n",
    "\n",
    "transposed_rds_sample10_df = rds_sample10_df.transpose()# Create an AnnData object\n",
    "adata10 = ad.AnnData(\n",
    "    X=transposed_rds_sample10_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample10_df.index},\n",
    "    var={'gene_id': transposed_rds_sample10_df.columns}  \n",
    ")\n",
    "transposed_rds_sample11_df = rds_sample11_df.transpose()# Create an AnnData object\n",
    "adata11 = ad.AnnData(\n",
    "    X=transposed_rds_sample11_df.values,  # Your data as numpy array\n",
    "    obs={'barcode': transposed_rds_sample11_df.index},\n",
    "    var={'gene_id': transposed_rds_sample11_df.columns}  \n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CAAGGACATCCCATCTCCGT', 'CAAGGACATCCCGGAGCTAA', 'CAAGGACATCCCGTGTTACT',\n",
       "       'CAAGGACATCCGACGCAGAA', 'CAAGGACATCCTATCTAGTG', 'CAAGGACATCCTCTTATTCG',\n",
       "       'CAAGGACATCGCACAGTATT', 'CAAGGACATCTACAACTAGC', 'CAAGGACATCTAGGTGGCGA',\n",
       "       'CAAGGACATCTAGTTAGGCT',\n",
       "       ...\n",
       "       'TTGTCTGCAAGCACAGTATT', 'TTGTCTGCAATACAACTAGC', 'TTGTCTGCAATAGGTGGCGA',\n",
       "       'TTGTCTGCAATAGTTAGGCT', 'TTGTCTGCAATCTGCAACAA', 'TTGTCTGCAATGCATCATAC',\n",
       "       'TTGTCTGCAATGGCCTTAAT', 'TTGTCTGCAATTACAACACG', 'TTGTCTGCAATTCAGGAGAT',\n",
       "       'TTGTCTGCAATTCTCTGATC'],\n",
       "      dtype='object', length=384)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "transposed_rds_sample11_df.index\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save h5ad files that are necessary for further Melisas pipeline\n",
    "adata1.write(h5ad_file_sample1)\n",
    "adata2.write(h5ad_file_sample2)\n",
    "adata3.write(h5ad_file_sample3)\n",
    "adata4.write(h5ad_file_sample4)\n",
    "adata5.write(h5ad_file_sample5)\n",
    "adata6.write(h5ad_file_sample6)\n",
    "adata7.write(h5ad_file_sample7)\n",
    "adata8.write(h5ad_file_sample8)\n",
    "adata9.write(h5ad_file_sample9)\n",
    "adata10.write(h5ad_file_sample10)\n",
    "adata11.write(h5ad_file_sample11)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(384, 35546)\n",
      "Campari\n",
      "Cre\n",
      "rRNA_18s_1870bp\n",
      "TdTomato\n",
      "EGFP\n",
      "iCre\n",
      "ReachR\n",
      "flp0\n"
     ]
    }
   ],
   "source": [
    "print(adata1.shape)\n",
    "genes_list = ['TdTomato','Cre','Campari','rRNA_18s_1870bp','EGFP','mNeptune','ReachR','flp0','iCre'] \n",
    "\n",
    "for gene in adata1.var['gene_id']:\n",
    "    for gene2 in genes_list : \n",
    "        if gene == gene2:\n",
    "            print(gene)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_names = []\n",
    "for i in range(adata1.shape[1]):\n",
    "    genes_names.append(adata1.var['gene_id'][i])\n",
    "adata1.var[\"gene_name\"] = genes_names\n",
    "adata1.var.index = adata1.var[\"gene_name\"]\n",
    "# make names unique\n",
    "adata1.var_names_make_unique()\n",
    "'Cre' in adata1.var['gene_name'] # check if Piezo1 is still there"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(384, 35546)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transposed_rds_sample1_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
